junit-platform-commons/src/main/java/org/junit/platform/commons/util/LruCache.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apiguardian.api.API;
+
+/**
+ * A simple LRU cache with a maximum size.
+ *
+ * @param <K> the type of keys maintained by this cache
+ * @param <V> the type of values maintained by this cache
+ */
+@API(status = INTERNAL, since = "1.6")
+public class LruCache<K, V> extends LinkedHashMap<K, V> {
+
+	private static final long serialVersionUID = 1L;
+
+	private int maxSize;
+
+	public LruCache(int maxSize) {
+		super(maxSize + 1, 0.75f, true);
```

Comment:
Why `maxSize + 1`?

Reply:
Because if the map already contains `maxSize` entries, it will add another one which will cause `HashMap` to resize and only then will it evict the eldest entry.

Reply:
OK. Thanks for clarifying.

However, if we really want to ensure that the map is not unnecessarily resized, I don't think adding 1 will suffice due to the `loadFactor`. A [little research](https://stackoverflow.com/a/6400874/388980) reveals that `maxSize * 4/3` is a more suitable `initialCapacity` to avoid resizing.

WDYT?

Reply:
I think the crucial sentence from `HashMap`'s Javadoc is this one:

> If the initial capacity is _greater_ than the maximum number of entries divided by the load factor, no rehash operations will ever occur.

Thus, if load factor is `.75`, it needs to be `(maxSize * 4/3f) + 1`.

Reply:
So, I think I found a good solution. Simply use a `loadFactor` of 1 and an `initialCapacity` of `maxSize + 1`. The new test shows that works reliably and is better than the solution above because it doesn't overestimate the maximum number of entries.

Reply:
OK. Let's stick with that approach. Assuming the hash codes returned for `java.lang.Class` result in an even distribution across the buckets, we shouldn't have collisions that result in slower performance.

I think we have now successfully over-engineered this feature. 🤓